rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -420,8 +425,33 @@ public boolean createNodeSource(@HeaderParam("sessionid") String sessionId,
                 }
             }
         }
-        return rm.createNodeSource(nodeSourceName, infrastructureType, infraParams, policyType, policyParams)
-                 .getBooleanValue();
+        try {
+            nsState.setResult(rm.createNodeSource(nodeSourceName,
+                                                  infrastructureType,
+                                                  infraParams,
+                                                  policyType,
+                                                  policyParams)
+                                .getBooleanValue());
+            return nsState;
```

Comment:
dont return here and in the catch, just add a finally block after and return it there.